### PR TITLE
[SPARK-10991][ML] logistic regression training summary handle empty prediction col

### PR DESCRIPTION
**Pull request by holdenk**
LogisticRegression training summary should still function if the predictionCol is set to an empty string or otherwise unset (related too https://issues.apache.org/jira/browse/SPARK-9718 )

### CODE REVIEW

**Comment by holdenk** [comment]
cc @jkbradley who asked for this change

**Comment by holdenk** [comment]
re-ping @jkbradley

**Comment by yu-iskw** [comment]
LGTM

**Comment by holdenk** [comment]
Hey @jkbradley or @dbtsai could you merge this if it looks ok to you as well? @yu-iskw took a look and the change is similar to the change we did as part of https://issues.apache.org/jira/browse/SPARK-9718 for the other model.

File: mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala [hunk 1]
```diff
@@ -434,6 +435,22 @@ class LogisticRegressionModel private[ml] (
...
+  private[classification] def findSummaryModelAndProbabilityCol():
+      (LogisticRegressionModel, String) = {
+    $(probabilityCol) match {
+      case "" =>
+        val probabilityColName = "probability_" + java.util.UUID.randomUUID.toString()
+        (copy(ParamMap.empty).setProbabilityCol(probabilityColName), probabilityColName)
+      case p => (this, p)
+    }
+  }
+
```
Comment (dbtsai) [inline mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala]: extra line

**Comment by dbtsai** [comment]
LGTM except the minor style.

**Comment by dbtsai** [comment]
Merged into master. Thanks.

**Pull request closed without merging**